controllers/compassmanager_controller.go

```diff
@@ -362,6 +361,45 @@ func (cm *CompassManagerReconciler) needsToBeReconciled(obj runtime.Object) bool
 	return false
 }
 
+func (cm *CompassManagerReconciler) needsToBeReconciledUpdate(oldObj, newObj runtime.Object) bool {
+	oldKymaObj, ok := oldObj.(*kyma.Kyma)
+	if !ok {
+		cm.Log.Error("Unexpected type detected. Old object is supposed to be of the Kyma type.")
```

Comment:
Could we maybe also print the current type of the object? Just to know better why the casting failed?